src/background/services/background.ts

```diff
@@ -154,28 +155,37 @@ export class Background {
   bindPermissionsHandler() {
     this.browser.permissions.onAdded.addListener(this.checkPermissions)
     this.browser.permissions.onRemoved.addListener(this.checkPermissions)
-    this.events.on('storage.host_permissions_update', async ({ status }) => {
-      this.logger.info('permission changed', { status })
+  }
+
+  bindStateHandler() {
+    this.events.on('storage.state_update', async ({ state, prevState }) => {
+      this.logger.info('state changed', { state, prevState })
       // TODO: change icon here in future
     })
   }
 
   bindOnInstalled() {
     this.browser.runtime.onInstalled.addListener(async (details) => {
-      this.logger.info(await this.storage.get())
+      const existingData = await this.storage.get()
+      this.logger.info(existingData)
       if (details.reason === 'install') {
         await this.storage.populate()
         await this.openPaymentsService.generateKeys()
       }
+      const migrated = await this.storage.migrate()
+      if (migrated) {
+        const prevVersion = existingData.version ?? 1
+        this.logger.info(`Migrated from ${prevVersion} to ${migrated.version}`)
```

Comment:
I know we are explicitly checking for the version in the `migrate` method to prevent migrating multiple times, but let's check for `details.reason === 'update'` as well.